### DISCUSSION TITLE
IGMPv3 supported?

### DISCUSSION THREAD

**Discussion by cdkeyleo** [troglobit/mcjoin#35] [Q&A] [answered]
Hello there, 

Does the tool support IGMPv3?

Cheers,

**Comment by troglobit** [comment, marked as answer]
It does not generate IGMP/MLD frames, the kernel does that. 

You provide an (*,G) or (S,G) and mcjoin sends a setsockopt() command to the kernel which, depending on the interface link, generates IGMP/MLD frames of a version that matches what other devices on your interface link have agreed on.
